central/sensor/service/pipeline/nodeinventory/pipeline_test.go

```diff
@@ -0,0 +1,100 @@
+package nodeinventory
+
+import (
+	"context"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	clusterDatastoreMocks "github.com/stackrox/rox/central/cluster/datastore/mocks"
+	nodeDatastoreMocks "github.com/stackrox/rox/central/node/datastore/mocks"
+	riskManagerMocks "github.com/stackrox/rox/central/risk/manager/mocks"
+	"github.com/stackrox/rox/central/sensor/service/common"
+	"github.com/stackrox/rox/generated/internalapi/central"
+	"github.com/stackrox/rox/generated/storage"
+	nodesEnricherMocks "github.com/stackrox/rox/pkg/nodes/enricher/mocks"
+)
+
+func Test_pipelineImpl_Run(t *testing.T) {
+	createMsg := func(id string) *central.MsgFromSensor {
+		return &central.MsgFromSensor{
+			Msg: &central.MsgFromSensor_Event{
+				Event: &central.SensorEvent{
+					Resource: &central.SensorEvent_NodeInventory{
+						NodeInventory: &storage.NodeInventory{
+							NodeId: id,
+						},
+					},
+				},
+			},
+		}
+	}
+	type mocks struct {
+		clusterStore  *clusterDatastoreMocks.MockDataStore
+		nodeDatastore *nodeDatastoreMocks.MockDataStore
+		riskManager   *riskManagerMocks.MockManager
+		enricher      *nodesEnricherMocks.MockNodeEnricher
+	}
+	type args struct {
+		ctx       context.Context
+		clusterID string
+		msg       *central.MsgFromSensor
+		injector  common.MessageInjector
+	}
+
+	tests := []struct {
+		name    string
+		mocks   mocks
+		args    args
+		wantErr string
```

Comment:
It looks like this could be a bool in this test, as the only thing we do with that is checking whether this is empty `tt.wantErr != ""`.

As another option, we could assert using:

```go 
s.ErrorContains(err, tt.wantErr)
```

Reply:
At [central/sensor/service/pipeline/nodeinventory/pipeline_test.go:53](https://github.com/stackrox/stackrox/pull/5145/files/9765951442ead153bc3455ae5590ba18b7af0a3f#diff-323ec1a5481b6d118c6a4ba909e66baa7f3f7fc469525cfb4bcb564abefd2a80R53) we check the error's text.

Reply:
I see that we do set it on line 53, but the only place we read it is in https://github.com/stackrox/stackrox/pull/5145/files/9765951442ead153bc3455ae5590ba18b7af0a3f#diff-323ec1a5481b6d118c6a4ba909e66baa7f3f7fc469525cfb4bcb564abefd2a80R95 where we only compare it against `""`.